pkg/diagnostics/diagnostics.go

```diff
@@ -49,10 +56,27 @@ func (r *Registry) RegisterConnectedPeersSource(
 	r.Registry.RegisterSource("connected_peers", func() string {
 		connectionManager := netProvider.ConnectionManager()
 		connectedPeers := connectionManager.ConnectedPeers()
+		connectedPeersAddrInfo := connectionManager.ConnectedPeersAddrInfo()
 
 		peersList := make([]map[string]interface{}, len(connectedPeers))
 		for i := 0; i < len(connectedPeers); i++ {
 			peer := connectedPeers[i]
+			peersAddrInfo := connectedPeersAddrInfo[i].Addrs
+
+			var peerAddr string
+			for _, peerAddrInfo := range peersAddrInfo {
+				// Peer may contain public and local addresses. We need to fetch a
+				// public address only. Range of local addresses starting from 127.* are
+				// skipped.
+				if strings.Contains(peerAddrInfo.String(), eligibleDNS4) ||
+					(strings.Contains(peerAddrInfo.String(), eligibleIP4) &&
+						!strings.Contains(peerAddrInfo.String(), "/127.")) {
+					// address is formatted as follows /<protocol_code>/<ip_address>/<connection_protocol>/<port>
+					// Address is  the 1st index
+					peerAddr = strings.Split(strings.Trim(peerAddrInfo.String(), "/"), "/")[1]
+				}
```

Comment:
Is it possible for a peer to bind to more than one address and one of them not being a loopback? 

I think I can bind to a public, externally available address as well as to an internal address that is not a loopback address (e.g. 10.0.0.0/8  or 172.16.0.0/12). Is it possible and if so, how is this handled?  I *think* this code may return 10.0.0.0/8 if it is first on the list.

Reply:
Good point, the current version filters out only ip4 IPs that start with `127.*`. It looks like the reserved "local" range of IPs are the following and we should add them as well:
- `10.*`
- `172.*`
- `192.*`
Thoughts?

Reply:
It would not be precise because each private address space has a different network mask: 10/8, 172.16/12, 192.168/16 so while 10/8 can be expressed as `10.*`, 192.168/16 can not be expressed as `192.*`. See [here](https://www.rfc-editor.org/rfc/rfc1918) please, section "3. Private Address Space".

I could also imagine the node can be configured with a private hostname, not accessible from the internet.

Let's take a step back. Why do we do this filtering in diagnostics at all? What is returned by `connectionManager.ConnectedPeersAddrInfo()` locally and what is returned by `connectionManager.ConnectedPeersAddrInfo()` on testnet?

Reply:
`connectionManager.ConnectedPeersAddrInfo()` on Goerli returns an array of peers with addresses. As of writing this comment, we have 7 peers
```
[
   {
      "16Uiu2HAmCcfVpHwfBKNFbQuhvGuFXHVLQ65gB4sJm7HyrcZuLttH":[
         /dns4/bootstrap-0.test.keep.network/tcp/3919
      ]
   }{
      "16Uiu2HAm3eJtyFKAttzJ85NLMromHuRg4yyum3CREMf6CHBBV6KY":[
         /dns4/bootstrap-1.test.keep.network/tcp/3919
      ]
   }{
      "16Uiu2HAmMosdpAuRSw1ahNhqFq8e3Y4d4c5WZkjW1FGQi5WJwWZ7":[
         /dns4/bst-a01.test.keep.boar.network/tcp/4001
      ]
   }{
      "16Uiu2HAmQaxEutkqzahSE19b9DhSXGCdb1BAfkGrt26obq1n6N9k":[
         /ip4/3.137.162.218/tcp/3919
      ]
   }{
      "16Uiu2HAky2Y4Tyq5vTA1CxikcDes6o5EH11i2qcg5dBV9W3Lks5c":[
         /ip4/34.141.9.57/tcp/3919
      ]
   }{
      "16Uiu2HAm77eSvRq5ioD4J8VFPkq3bJHBEHkssCuiFkgAoABwjo2S":[
         /dns4/keep-validator-0.eks-ap-northeast-2-secure.staging.staked.cloud/tcp/3919
      ]
   }{
      "16Uiu2HAm1buBFu8sHmgToezz6994uqQCZZ42azxrJP5GTxWmcBPA":[
         /ip4/138.68.240.252/tcp/3919
      ]
   }
]
```

Reply:
OK, and then why do we need to filter them out? Isn't it that local addresses are returned only for local deployments which is totally fine?

Reply:
Looking at this response we do not. If we can assume that there would be no local addresses in the response on the test / mainnet then we can remove this filter. I was leaning towards "it won't hurt" adding an extra "if" in case I don't know something about the libp2p lib.